go-controller/pkg/allocator/udn/ips_test.go

```diff
@@ -0,0 +1,64 @@
+package udn
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"k8s.io/utils/ptr"
+)
+
+func TestAllocateMasqueradeIPs(t *testing.T) {
+	type masqueradeIPs struct {
+		Shared, Local string
+	}
+	var testCases = []struct {
+		description   string
+		networkID     int
+		subnet        string
+		expectedError *string
+		expectedIPs   masqueradeIPs
+	}{
+		{
+			description: "with proper network id 2 should return expected subnets",
+			networkID:   2,
+			subnet:      "169.254.169.0/19",
+			expectedIPs: masqueradeIPs{Shared: "169.254.169.13", Local: "169.254.169.14"},
```

Comment:
If we start from offset = 0 like commented above this will generate 169.254.168.12 (base + network-id) which might make it easier to debug.

Reply:
answered above too.